File: BKStephens/euler/four.exs
defmodule Four do
  def calculate, do: palindrome_products(100, 999) |> Enum.max
  
  def palindrome_products(min, max) do
    for x <- min..max,
        y <- x..max,
        is_palindrome?(x * y),
        do: [x * y]
  end

  def is_palindrome?(n), do: to_string(n) == String.reverse(to_string(n))
end

File: BKStephens/euler/four.rb
class Four 
  def self.calculate
    min, max = 100, 999
    max_palindrome = 0

    (min..max).each do |x|
      (x..max).each do |y|
        product = x * y
        max_palindrome = product if product > max_palindrome && is_palindrome?(product)
      end
    end
     
    max_palindrome
  end
  
  def self.is_palindrome?(n)
    n.to_s == n.to_s.reverse
  end
end
